src/ui/atoms/textField/TextField.tsx

```diff
@@ -20,44 +20,24 @@ export type TextFieldProps = InputBaseProps & {
    * If true, the TextField will take 100% of its parent's size.
    */
   readonly fullWidth?: boolean
-  /**
-   * If true, the TextField allows multiline text as a text area.
-   */
-  readonly multiline?: boolean
-  /**
-   * If multiline, defines the number of lines of the text area.
-   */
-  readonly numberOfLines?: number
-  /**
-   * If true and multiline, the user can't resize the text area.
-   */
-  readonly disableAreaResize?: boolean
 }
 
 export const TextField: React.FC<TextFieldProps> = ({
   size = 'medium',
   helperText,
   fullWidth = false,
-  multiline = false,
-  numberOfLines,
-  disableAreaResize = false,
   ...props
 }: TextFieldProps): JSX.Element => {
   const containerClass = classNames(
-    `okp4-text-field-main ${multiline && !disableAreaResize ? 'auto' : size}`,
+    `okp4-text-field-main ${props.multiline && !props.disableAreaResize ? 'auto' : size}`,
     {
-      'full-width': !multiline && fullWidth
+      'full-width': !props.multiline && fullWidth
     }
   )
 
   return (
     <div className={containerClass}>
-      <InputBase
-        disableAreaResize={disableAreaResize}
-        multiline={multiline}
-        numberOfLines={numberOfLines}
-        {...props}
-      />
+      <InputBase {...props} />
```

Comment:
I can't see clearly how this refactor is linked to the 'add border' feature. 
I'll suggest to open a dedicated pull request to handle this changes and then limit this one to the incoming feature 😌